packages/utils/src/utils/bytecode.ts

```diff
@@ -0,0 +1,20 @@
+import type { BytesLike } from '@fuel-ts/interfaces';
+import { gzipSync, gunzipSync } from 'fflate';
```

Comment:
Will this `fflate` lib work in browser envs?

How much kbs does it add to the final bundle?

Reply:
Yes, I have ensured this by running the unit tests for these functions in the browser environment as well.

Reply:
Great.

Do you know how much kbs it adds to the final bundle?

It may be valuable information.

Reply:
I'm not sure, investigating.

Reply:
Bundle size comparisons:

Before: 
```
308K	dist/main.js
 92K	dist/main.js.gz
 ```
 
After:
```
316K	dist/main.js
 96K	dist/main.js.gz
 ```

Reply:
We'll be adding 4 kB to _all_ applications using our sdk, regardless if they're deploying contracts or not. Given that contracts are deployed once and then just interacted with in a production app, it seems to me that it's wiser to leave things as they are. It's the users' fault if they add the contract bytecode into their apps even though the contract is already deployed and they can interact with it via just its id and abi.

Reply:
In the case of contracts, the bytecode is optional.

But it is required for scripts and predicates.

Reply:
Should we consider using native modules for this task?

NodeJS refs:
 - https://stackoverflow.com/a/63007331

Browser refs:
 - https://stackoverflow.com/a/247261
 - https://stackoverflow.com/a/68829631

This API needs to be built like the crypto package, in a hybrid way targeting NodeJS + Browser, using native modules of each. I guess this could reduce the size increase to almost zero.

Reply:
@arboleya that sounds fair. Do you reckon we'll need to create a new package for this then since we'll need two different bundles for the two environments? Or should I just tweak the `utils` package?

Reply:
I guess you could keep the utility in the `utils` package and configure it like the `crypto` so you can have individual entry points for NodeJS and the Browser. cc @Torres-ssf

Reply:
@arboleya I looked into this and while I found a straightforward solution for node, unfortunately there doesn't seem to be any way to do the compression in browser using Native APIs without it being async. This won't work for our use case.

We could re-write the whole algorithm ourselves just for the browser but that can turn out to be tricky. It's not exactly straightforward as it involves two or three different 'steps', and even then our custom rewrite might not turn out to be that efficient.

I think we should take the 4kb hit and go with the external library.